### PR TITLE
add player names to process for creating games via mobile

### LINKED ISSUE #304: write test for: "create from mobile" functionality

### LINKED ISSUE #447: add player names to process of creating games via mobile

### PR DESCRIPTION
**Pull request by tongxiang**
#### What's this PR do?
We've added functionality to the responder which uses player names in the onboarding process and to end-level feedback. 

Before, we were only able to create SMS games with player names through the webform. This PR: 

1. adds functionality to allow SMS games with player names to be created through by texting in to a keyword (the mobile-create game creation process.) Note that we've added two new opt in paths to the game config model. 
2. and adds tests. 

#### Where should the reviewer start?
`SGCreateFromMobileController.js`. Note that the main function has been refactored so that it no longer has a third link in the promise chain, and no longer requires a custom error object thrown to break the promise chain. 

New test file with `CreateGameFromMobile.js`.

#### How should this be manually tested?
`npm test`. 

Additionally, you can use the below postman collection to test the mobile-create onboarding process. Use the `Create From Mobile` request set up to first: 

1. text in the alpha name. 
2. text in beta 1's name and number, i.e. "Kenia 5555551000". 
3. text in beta 2's name and number. 
4. text in beta 3's name and number. 

If the alpha fails to provide a name for herself, or a name or number for each of the betas, the responder will send her an error message. `npm test` tests these cases. 

```
{"id":"b533f5f3-c898-6e2b-e339-0650d3aba28d","name":"SMS Create From Mobile","timestamp":1411504417956,"requests":[{"collectionId":"b533f5f3-c898-6e2b-e339-0650d3aba28d","id":"4be69804-caad-f0ef-cc11-4c9e12c48da1","name":"Create From Mobile","description":"First send an arg as a friend's phone number, then text back Y","url":"http://localhost:4711/sms-multiplayer-game/mobile-create?story_type=competitive-story&story_id=301","method":"POST","headers":"","data":[{"key":"story_id","value":"301","type":"text"},{"key":"story_type","value":"competitive-story","type":"text"},{"key":"phone","value":"15555550000","type":"text"},{"key":"args","value":"15555550001","type":"text"}],"dataMode":"urlencoded","timestamp":0,"version":2,"time":1431545915184},{"collectionId":"b533f5f3-c898-6e2b-e339-0650d3aba28d","id":"563c7839-74a4-cfa9-ae8e-bfb4bfdfc1f2","name":"Ariel Game Action","description":"","url":"http://localhost:4711/sms-multiplayer-game/user-action?story_type=competitive-story","method":"POST","headers":"","data":[{"key":"phone","value":"15555550001","type":"text"},{"key":"args","value":"A","type":"text"}],"dataMode":"urlencoded","timestamp":0,"version":2,"time":1431118082786},{"collectionId":"b533f5f3-c898-6e2b-e339-0650d3aba28d","id":"6b996b66-19a6-dfd8-bffe-6fa7d4cf2a90","name":"Tong Game Action","description":"","url":"http://localhost:4711/sms-multiplayer-game/user-action?story_type=competitive-story","method":"POST","headers":"","data":[{"key":"phone","value":"15555550000","type":"text"},{"key":"args","value":"A","type":"text"}],"dataMode":"urlencoded","timestamp":0,"version":2,"time":1431545931301},{"collectionId":"b533f5f3-c898-6e2b-e339-0650d3aba28d","id":"d031afa4-4330-f629-b744-916694bf7de9","name":"Invite Friends Now","description":"texts back Y","url":"http://localhost:4711/sms-multiplayer-game/mobile-create?story_type=competitive-story&story_id=101","method":"POST","headers":"","data":[{"key":"story_id","value":"101","type":"text"},{"key":"story_type","value":"competitive-story","type":"text"},{"key":"phone","value":"15555550000","type":"text"},{"key":"args","value":"Y","type":"text"}],"dataMode":"urlencoded","timestamp":0,"version":2,"time":1431545927943},{"collectionId":"b533f5f3-c898-6e2b-e339-0650d3aba28d","id":"ed717c23-0077-50b4-8055-49b5445dc6ca","name":"Ariel is the Beta Joining","description":"","url":"http://localhost:4711/sms-multiplayer-game/beta-join?story_type=competitive-story","method":"POST","headers":"","data":[{"key":"phone","value":"15555550001","type":"text"},{"key":"args","value":"y","type":"text"}],"dataMode":"urlencoded","timestamp":0,"version":2,"time":1431118035054}]}
```

#### Any background context you want to provide?
The new onboarding process: 
1) alpha texts in a keyword, i.e. `BULLY`
2) alpha texts in her first name
3) alpha texts in a beta's 1's first name and number
4) alpha texts in a beta's 2's first name and number
5) alpha texts in a beta's 3's first name and number

#### What are the relevant tickets?
Closes #304, closes #447.

### CODE REVIEW

File: app/lib/string-validators/index.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ function isValidEmail(email) {
...
...bert piercing', 'pthc', 'pubes', 'pussy', 'queaf', 'raghead', 'ragingboner', 'rape', 'raping', 'rapist', 'rectum', 'reversecowgirl', 'rimjob', 'rimming', 'rosypalm', 'rosypalm and her 5 sisters', 'rustytrombone', 'sadism', 'scat', 'schlong', 'scissoring', 'semen', 'sex', 'sexo', 'sexy', 'shavedbeaver', 'shavedpussy', 'shemale', 'shibari', 'shit', 'shota', 'shrimping', 'slanteye', 'slut', 's&m', 'smut', 'snatch', 'snowballing', 'sodomize', 'sodomy', 'spic', 'spooge', 'spreadlegs', 'strapon', 'strapon', 'strappado', 'stripclub', 'styledoggy', 'suck', 'sucks', 'suicidegirls', 'sultrywomen', 'swastika', 'swinger', 'taintedlove', 'tastemy', 'teabagging', 'threesome', 'throating', 'tiedup', 'tightwhite', 'tit', 'tits', 'titties', 'titty', 'tongueina', 'topless', 'tosser', 'towelhead', 'tranny', 'tribadism', 'tubgirl', 'tubgirl', 'tushy', 'twat', 'twink', 'twinkie', 'twogirls one cup', 'undressing', 'upskirt', 'urethraplay', 'urophilia', 'vagina', 'venusmound', 'vibrator', 'violetblue', 'violetwand', 'vorarephilia', 'voyeur', 'vulva', 'wank', 'wetback', 'wetdream', 'whitepower', 'womenrapping', 'wrappingmen', 'wrinkledstarfish', 'xx', 'xxx', 'yaoi', 'yellowshowers', 'yiffy', 'zoophilia' ]
```
Comment (jonuy) [inline app/lib/string-validators/index.js]: I see `ass` got removed from the naughty list :innocent:

Comment (tongxiang) [inline app/lib/string-validators/index.js]: hahaha names like "Cassie" would've gotten flagged, and I didn't feel like writing a super complex regex to account for those situations.

File: test/CreateGameFromMobile.js [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+      query: {
+        story_id: storyId,
+        story_type: storyType
+      },
+      body: {
+        phone: alphaPhone,
+        args: betaName1 + betaPhone1
+      }
+    }
+    it('should create a game-create-config doc updated with ' + betaName0 + '\'s name and number', function(done) {
```
Comment (jonuy) [inline test/CreateGameFromMobile.js]: Should be `betaName1`?

Comment (tongxiang) [inline test/CreateGameFromMobile.js]: Yup yup thanks!

**Comment by jonuy** [comment]
I spent time messing around with the test and trying out different combos of names and phone numbers, like adding spaces, removing the leading '1' in the phone numbers, throwing in other common phone number characters like `-`, `(`, `)`, and `.`.  All seemed to work out well. :+1:

**Pull request merged**